magic_spoiler/__main__.py

```diff
@@ -456,8 +456,8 @@ def compare_json_content(f1: str, f2: str) -> bool:
     file2 = pathlib.Path(f2)
 
     if file1.is_file() and file2.is_file():
-        f1_hash = hashlib.sha512(file1.open("r").read().encode()).hexdigest()
-        f2_hash = hashlib.sha512(file2.open("r").read().encode()).hexdigest()
+        f1_hash = hashlib.sha512(file1.open("rb").read().encode()).hexdigest()
+        f2_hash = hashlib.sha512(file2.open("rb").read().encode()).hexdigest()
```

Comment:
>I've adapted the script to require utf-8

Even better, that's great!


However, there is a new issue now with `.encode()`:
```
"C:\...\GitHub\Cockatrice\Magic-Spoiler>python -m magic_spoiler
Downloaded: https://api.scryfall.com/sets/ (Cache = False)
Handling J21
Downloaded: https://api.scryfall.com/sets/J21 (Cache = False)
Downloaded: https://api.scryfall.com/cards/search?order=set&q=e%3Aj21&unique=prints (Cache = False)
No new data in J21.xml, skipping replacement
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.1520.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.9_3.9.1520.0_x64__qbz5n2kfra8p0\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "C:\...\GitHub\Cockatrice\Magic-Spoiler\magic_spoiler\__main__.py", line 657, in <module>
    main()
  File "C:\...\GitHub\Cockatrice\Magic-Spoiler\magic_spoiler\__main__.py", line 636, in main
    changed |= write_set_json(trice_dict, set_info)
  File "C:\...\GitHub\Cockatrice\Magic-Spoiler\magic_spoiler\__main__.py", line 549, in write_set_json
    if compare_json_content(str(output_file_path), old_xml_location):
  File "C:\...\GitHub\Cockatrice\Magic-Spoiler\magic_spoiler\__main__.py", line 459, in compare_json_content
    f1_hash = hashlib.sha512(file1.open("rb").read().encode()).hexdigest()
AttributeError: 'bytes' object has no attribute 'encode'
```

Reply:
encoding is no longer necessary, it's reading in binary mode now
```suggestion
        f1_hash = hashlib.sha512(file1.open("rb").read()).hexdigest()
        f2_hash = hashlib.sha512(file2.open("rb").read()).hexdigest()
```